enhancements/security/openshift-image-policy.md

```diff
@@ -0,0 +1,359 @@
+It's possible that an ImageDigestMirrorSet like this would be sufficient:
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: ImageDigestMirrorSet
+metadata:
+  name: openshift-release-payload
+spec:
+  imageDigestMirrors:
+  - mirrors:
+    - example.com/my-mirror/whatever
+    source: quay.io/openshift-release-dev/ocp-release
+```
+
+In any case, it's out of scope for a reason, but until it gets picked up, it might make things more difficult for `TechPreviewNoUpgrade` disconnected/restricted-network clusters.
+
+##### Mirror registries that do not support Sigstore signatures
+
+As a subset of the restricted-network use case, when a user has an ImageDigestMirrorSet set to point at a mirror, and the canonical quay.io is either not reachable, or [`mirrorSourcePolicy: NeverContactSource`][ImageDigestMirrorSet] is set, the mirror registry will need to allow Sigstore signatures in order for CRI-O to be able to retrieve signatures.
+Quay support for Sigstore is [new and ongoing][PROJQUAY-3400], so it's likely that there are other mirror registries in use where Sigstore hosting is not yet possible.
+But again, restricted-network use is out of scope for this initial policy, and once it comes into scope, mirror support will be a generic issue for all ClusterImageProfile and ImageProfile users, and will not be unique to OpenShift's internal policies.
+
+#### Security
+
+How will security be reviewed and by whom?
+
+I have no idea; suggestions welcome :).
+
+There will be a gap during install before the ClusterImagePolicy has been created and rolled out to nodes, during which unsigned `quay.io/openshift-release-dev/ocp-release` images might run.
+But we do not perform any in-cluster signature checking during installs today; they are secured by a trusted installer using a baked-in release image digest.
+Trust in the installer is either transitive via trust in the hosting site from which the installer was downloaded, or via mirror signatures like [this one][4.15.0-multi-mirror-signature].
+
+#### User experience
+
+How will UX be reviewed and by whom?
+
+I expect "images are failing to meet the relevant image policy/policies" to be something with a solid user experience, regardless of whether OpenShift defines its own image policy, or leaves image policies entirely up to users.
+
+### Drawbacks
+
+We're building a new thing here without a lot of experience defining ClusterImagePolicies.
+And not any for the `quay.io/openshift-release-dev/ocp-release` scope, because those don't have Sigstore signatures [yet][OTA-1267].
+Designing core-image verification without much experience is risky.
+But passing around OpenPGP / GPG signatures in non-registry storage for disconnected/restricted-networks is awkward, so doing nothing has risks too.
+
+Building something that is partly designed to make mirroring and disconnected/restricted-network use easier without understanding what the disconnected/restricted-network solution will look like is also risky.
```

Comment:
With the drawbacks listed like this, the "minimal step forward that seems to make sense" approach you've chosen for this enhancement seems to be the right approach to me.

Reply:
I'm going to mark this thread resolved, but if you think it should lead to changed Markdown in the enhancement, let me know :).